lib/inertia_rails/renderer.rb

```diff
@@ -112,5 +115,32 @@ def resolve_component(component)
 
       configuration.component_path_resolver(path: controller.controller_path, action: controller.action_name)
     end
+
+    def validate_partial_props(merged_props)
+      return unless rendering_partial_component?
+
+      unrequested_props = merged_props.select do |key, prop|
+        !key.in?(partial_keys)
```

Comment:
This code might be called on every Inertia request, so I prefer optimized code over readability. For example, we can eliminate the second loop and avoid creating an additional array by using:
```suggestion
        !key.in?(partial_keys) && !prop.respond_to?(:call)
```

Reply:
I shudder to think of the application with an action with so many props that this would make a difference 😅, but this is easy enough to change.

You guessed correctly that it's for readability.